### PR TITLE
Support Appliance Network Interface

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Support Appliance Network Interface

### Check List
- [ ] New functionality includes sanity testing.
  - [ ] All sanity tests pass. (`$ ansible-test sanity`).
  - [ ] All unit tests pass.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).

### CODE REVIEW

File: plugins/modules/oneview_appliance_network_interface_facts.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    )
+
+    def __init__(self):
+        super().__init__(additional_arg_spec=self.argument_spec)
+        self.set_resource_object(self.oneview_client.appliance_network_interfaces)
+
+    def execute_module(self):
+        if self.module.params.get('mac_address'):
+            network_interface = self.resource_client.get_by_mac_address(self.module.params.get('mac_address'))
+        else:
```
Comment (VenkateshRavula) [inline plugins/modules/oneview_appliance_network_interface_facts.py:85]: Add the facts of ```get_all_mac_address``` also

File: plugins/modules/oneview_appliance_network_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+from ansible_collections.hpe.oneview.plugins.module_utils.oneview import OneViewModule, dict_merge, compare
+
+
+class ApplianceNetworkInterfacesModule(OneViewModule):
+    MSG_CREATED = 'Appliance Network Interface created successfully.'
+    MSG_ALREADY_PRESENT = 'Appliance Proxy Configuration is already present.'
+
+    def __init__(self):
+        argument_spec = dict(
+            data=dict(required=False, type='dict'),
```
Comment (VenkateshRavula) [inline plugins/modules/oneview_appliance_network_interfaces.py:97]: make ```data``` parameter as mandatory.

File: roles/oneview_appliance_network_interface_facts/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- debug: var=appliance_network_interfaces
+
+- name: Gather facts about the Network Interfaces by Mac Address
+  oneview_appliance_network_interface_facts:
+    config: "{{ config }}"
+    params:
+       mac_address: "{{ mac_address }}"
+  delegate_to: localhost
+
+- debug: var=appliance_network_interfaces
```
Comment (VenkateshRavula) [inline roles/oneview_appliance_network_interface_facts/tasks/main.yml:16]: Add a task to get facts about all mac address

File: roles/oneview_appliance_network_interfaces/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+         - "{{ dnsServer2 }}"
+  delegate_to: localhost
+- debug: var=appliance_network_interfaces
+
+- name: Do nothing when network interface exists
+  oneview_appliance_network_interfaces:
+    config: "{{ config }}"
+    state: present
+    data:
+      macAddress: "{{mac_address}}"
```
Comment (VenkateshRavula) [inline roles/oneview_appliance_network_interfaces/tasks/main.yml:27]: try with entire data to check the corner idempotency issues

Comment (yuvirani) [reply roles/oneview_appliance_network_interfaces/tasks/main.yml:27]: I have already tested with all the params above for idempotency issues. I gave mac_address just to keep task information minimal.

Comment (VenkateshRavula) [reply roles/oneview_appliance_network_interfaces/tasks/main.yml:27]: ok. But please add entire data in example.

**Review by VenkateshRavula — APPROVED** [review]
LGTM

**Review by chebroluharika — APPROVED** [review]

**Pull request merged**